modules/internal/localeModels/flat/LocaleModel.chpl

```diff
@@ -395,20 +400,14 @@ module LocaleModel {
     const node = chpl_nodeFromLocaleID(loc);
     if (node == chpl_nodeID) {
       if __primitive("task_get_serial") then
-        // don't call the runtime nb execute_on function if we can stay local
         chpl_ftable_call(fn, args);
       else
-        // We'd like to use a begin, but unfortunately doing so as
-        // follows does not compile for --no-local:
-        // begin chpl_ftable_call(fn, args);
-        chpl_comm_execute_on_nb(node, chpl_sublocFromLocaleID(loc),
-                          fn, args, args_size);
+        chpl_task_taskCallFTable(fn, args, args_size, c_sublocid_any);
     } else {
-      const subloc = chpl_sublocFromLocaleID(loc);
       if __primitive("task_get_serial") then
-        chpl_comm_execute_on(node, subloc, fn, args, args_size);
+        chpl_comm_execute_on(node, c_sublocid_any, fn, args, args_size);
       else
-        chpl_comm_execute_on_nb(node, subloc, fn, args, args_size);
+        chpl_comm_execute_on_nb(node, c_sublocid_any, fn, args, args_size);
```

Comment:
Nope, I was wrong.  It's better to use 'any' here.  There are checks in the runtime about not using 'none' in places where it would be hard to do the check differently for flat and numa.